tests/integration/feature_repos/test_repo_configuration.py

```diff
@@ -0,0 +1,434 @@
+
+    This feature store object can be interacted for the purposes of tests.
+    The user is *not* expected to perform any clean up actions.
+
+    :param test_suite_name: A name for the test suite.
+    :param test_repo_config: configuration
+    :param stop_at_step: The step which should be the last one executed when setting up the test environment.
+    :param data_source_cache:
+    :return: A feature store built using the supplied configuration.
+    """
+    with construct_test_environment(test_suite_name, test_repo_config) as environment:
+        fs = environment.feature_store
+        fvs = []
+        entities = []
+        try:
+            if stop_at_step >= EnvironmentSetupSteps.CREATE_OBJECTS:
+                if data_source_cache is not None:
+                    fixtures = data_source_cache.get(
+                        test_repo_config.offline_store_creator, None
+                    )
+                    if fixtures:
+                        environment = setup_entities(
+                            environment, entities_override=fixtures[0]
+                        )
+                        environment = setup_datasets(
+                            environment, datasets_override=fixtures[1]
+                        )
+                        environment = setup_data_sources(
+                            environment, data_sources_override=fixtures[2]
+                        )
+                    else:
+                        environment = setup_entities(environment)
+                        environment = setup_datasets(environment)
+                        environment = setup_data_sources(environment)
+                        data_source_cache[test_repo_config.offline_store_creator] = (
+                            environment.entities,
+                            environment.datasets,
+                            environment.datasources,
+                            environment.data_source_creator,
+                        )
+                else:
+                    environment = setup_entities(environment)
+                    environment = setup_datasets(environment)
+                    environment = setup_data_sources(environment)
+
+                environment = setup_feature_views(environment)
+            if stop_at_step >= EnvironmentSetupSteps.APPLY_OBJECTS:
+                entities.extend([driver(), customer()])
+                fvs.extend(environment.feature_views.values())
+                fs.apply(fvs + entities)
+            if stop_at_step >= EnvironmentSetupSteps.MATERIALIZE:
+                fs.materialize(environment.start_date, environment.end_date)
+
+            yield environment
+        finally:
+            if (
+                data_source_cache is None
+                and stop_at_step >= EnvironmentSetupSteps.CREATE_OBJECTS
+            ):
+                environment.data_source_creator.teardown()
+
+
+def parametrize_offline_retrieval_test(offline_retrieval_test):
```

Comment:
I made a similar comment above at `vary`, but it's not clear to me why we need these `parametrize_x` functions. We already have parameterization in pytest, and if it's about dynamic marking it's pretty straightforward

```python
import pytest


@pytest.mark.parametrize(
    ("n", "expected"),
    [
        pytest.param(1, 3, marks=[pytest.mark.local, pytest.mark.integration]),
        pytest.param(1, 3, marks=[pytest.mark.aws, pytest.mark.integration]),
    ],
)
def test_increment(n, expected):
    assert n + 1 == expected
```

I just want to make sure it's 100% obvious by looking at a test file what is happening. The more we abstract away the test cases from the test code the slower it is to understand and develop tests.